MetaCell/salk-interactive-atlas
# Feature/256 - Rename populations from the web application interface

Closes #256 

Implemented solution: 
- Created APIs for rename_population and rename_subpopulation. Check for authorization, if subpopulation exists in experiment, is not residential population.
- Created API spec from BE -> yaml -> FE. Called - ExperimentRenamePopulations, ExperimentRenameSubPopulations.
- Frontend API integration. On double click go into edit mode. On Enter submits the request to update the population/subpopulation name.


How to test this PR: 
- Open an experiment
- Double click any one of the experimental population
- Change the name 
- Press Enter to submit the request
- Check changes implemented after refresh. 

Video description:

https://github.com/MetaCell/salk-interactive-atlas/assets/59233227/ccfddb33-ad5f-47ad-90de-704b8cd2693f





### Sanity checks:

### Breaking changes (select one):

### Possible deployment updates issues (select one):

### Test coverage (select one):

### Documentation (select one):

### Nice to have (if relevant):
